packages/snaps-controllers/src/types/controllers.ts

```diff
@@ -102,7 +102,7 @@ export type TransactionParams = {
   /**
    * Unique number to prevent replay attacks.
    */
-  nonce?: string;
+  nonce?: number | string;
```

Comment:
We use `string` internally to represent nonces, so we shouldn't change this to allow numbers.

```suggestion
  nonce?: string;
```